File: tests/rosettacode/matrix.go
package main

import "fmt"

func transpose(m [][]int) [][]int {
	rows := len(m)
	cols := len(m[0])
	t := make([][]int, cols)
	for i := range t {
		t[i] = make([]int, rows)
		for j := range t[i] {
			t[i][j] = m[j][i]
		}
	}
	return t
}

func printMatrix(m [][]int) {
	for _, row := range m {
		for j, v := range row {
			if j > 0 {
				fmt.Print(" ")
			}
			fmt.Printf("%2d", v)
		}
		fmt.Println()
	}
}

func main() {
	m := [][]int{
		{1, 2, 3, 4},
		{5, 6, 7, 8},
		{9, 10, 11, 12},
	}
	fmt.Println("Original:")
	printMatrix(m)
	fmt.Println("Transposed:")
	printMatrix(transpose(m))
}

File: tests/java-rosettacode/Matrix.java
public class Matrix {
    public static int[][] multiply(int[][] a, int[][] b) {
        int rows = a.length;
        int cols = b[0].length;
        int inner = b.length;
        int[][] result = new int[rows][cols];
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                for (int k = 0; k < inner; k++) {
                    result[i][j] += a[i][k] * b[k][j];
                }
            }
        }
        return result;
    }

    public static void printMatrix(String label, int[][] m) {
        System.out.println(label);
        for (int[] row : m) {
            System.out.printf("  [%d, %d]%n", row[0], row[1]);
        }
    }

    public static void main(String[] args) {
        int[][] a = {{1, 2}, {3, 4}};
        int[][] b = {{5, 6}, {7, 8}};
        int[][] c = multiply(a, b);
        printMatrix("A:", a);
        printMatrix("B:", b);
        printMatrix("A * B:", c);
    }
}
